### PR TITLE
Bug/export directory create error

### LINKED ISSUE #10: export directory create error
### Search before asking

- [X] I have searched the [issues](https://github.com///issues) and found no similar bug report.

### Select Component

_No response_

### Bug

It need to create directory when export dataset.

### example
- when raw data structure is like this
```
raw/
  a/ 
    b/
```
need to create `a/b` under `exports/{format}/{set}/{images} and {labels}`.

### Environment

_No response_

### Minimal Reproducible Example

_No response_

### Additional

_No response_

### PR DESCRIPTION
**Pull request by ZeroAct**
- make directory if raw data path has child directories.
- add strict option to type_validator
  - if use `@type_validator(float, strict=False)`, it will automatically cast type to float.

### CODE REVIEW

**Review by dustjr1565 — APPROVED** [review]

File: waffle_utils/dataset/fields/annotation.py [hunk 1]
```diff
@@ -12,10 +12,10 @@ def __init__(
         img_id: int,
         # optional
         cat_id: int = None,
-        bbox: list[int] = None,
+        bbox: list[float] = None,
```
Comment (echoyunjin) [inline waffle_utils/dataset/fields/annotation.py:15]: Why did you change from int to float?

**Review by echoyunjin — APPROVED** [review]

File: waffle_utils/dataset/fields/annotation.py (hunk 1 above)
Comment (ZeroAct) [reply waffle_utils/dataset/fields/annotation.py:15]: I didn't know the types of coco format, so I did set the format to int with my common sense.
But, it was float type. 😂
